src/context.cpp

```diff
@@ -640,9 +643,11 @@ boost::optional<const actor_t&>
 context_t::locate(const std::string& name) const {
     auto ptr = m_services.synchronize();
     auto it  = std::find_if(ptr->begin(), ptr->end(), match{name});
-
+    if(it == ptr->end()) {
+        return boost::none;
+    }
     return boost::optional<const actor_t&>(
-        it != ptr->end() && it->second->is_active(),
```

Comment:
Also, what was the point of breaking `it != ptr->end() && it->second->is_active()` into 2 parts? How does that fix a segmentation fault?